pkg/oauth/registry/oauthaccesstoken/strategy.go

```diff
@@ -56,8 +58,13 @@ func (s strategy) Validate(ctx apirequest.Context, obj runtime.Object) field.Err
 		return append(validationErrors, field.InternalError(field.NewPath("clientName"), err))
 	}
 	if err := scopeauthorizer.ValidateScopeRestrictions(client, token.Scopes...); err != nil {
-		return append(validationErrors, field.InternalError(field.NewPath("clientName"), err))
+		validationErrors = append(validationErrors, field.InternalError(field.NewPath("clientName"), err))
 	}
+	if timeout := client.AccessTokenTimeoutSeconds; timeout != nil {
+		if *timeout != token.TimeoutsIn {
+			validationErrors = append(validationErrors, field.Invalid(field.NewPath("timeoutsIn"), token.TimeoutsIn, fmt.Sprintf("must equal %d", *timeout)))
+		}
+	} // TODO technically we should have an else if with the default timeout from master config
```

Comment:
I think the only validation here should be that timeout is not negative, ther may be reasons in future to change the initial timeout, and this does not validate the large parte of timeouts we get ina  default setting when the timeout is defined only by master config anyway.

Reply:
I made it validate based on both the default and the OAuth client timeout, but only if there is a default to indicate the feature is on.  We can always loosen validation in the future, but we cannot tighten it (considered an API change, break `oc adm migrate`, etc).  Thus I wanted to have the tightest validation possible to start with, and loosen if a valid scenario warrants it.